src/mango_native_proc.erl

```diff
@@ -210,6 +218,13 @@ get_default_enabled(Props) ->
             couch_util:get_value(<<"enabled">>, Opts, true)
     end.
 
+get_array_length_enabled(Props) ->
+    case couch_util:get_value(<<"array_length_field">>, Props, {[]}) of
```

Comment:
I see you've followed the above method but it's not a good model. array_length_field is going to be either true or false, right? so `couch_util:get_value(<<"array_length_field">>, Props, true).` is all you need.